Validator/src/main/java/org/openzen/zenscript/validator/visitors/ExpressionValidator.java

```diff
@@ -687,6 +687,10 @@ private void checkCorrectType(CodePosition position, TypeID expected, TypeID act
 		if (actual == BasicTypeID.INVALID)
 			return; // in this case the underlying error will already by reported elsewhere
 
+		if(expected.isOptional() && actual == BasicTypeID.NULL) {
```

Comment:
Question: Is this change here correct, or should it have already been casted to optional by here?